small changes, mostly with types

subtosharki opened the pull request:
- added explicit type import
- added missing alt tags
- simplified imports
- removed unused imports
- removed unused code
- added type for settings in index.ts instead of it being any
- in load-instances.ts switched from forEach to for-of because each promise returned was ignored
- added await where needed
- added ide specific folders to .gitignore
- simplified code in areas

subtosharki committed:
- added explicit type import
- added missing alt tags
- simplified imports
- removed unused imports
- removed unused code
- added type for settings in index.ts instead of it being any
- in load-instances.ts switched from forEach to for-of because each promise returned was ignored
- added await where needed
- added ide specific folders to .gitignore
- simplified code in areas

whitigol commented on .gitignore:
```diff
@@ -39,6 +39,8 @@ yarn-error.log
 .DS_Store
 *.pem
+.idea
+.vscode
```
```suggestion
.idea
```
Keeping `.vscode` for recommended vscode extensions 🙂

whitigol commented on apps/api/src/index.ts:
```diff
@@ -106,14 +106,14 @@ async function initAPI() {
     // #region Start Server
-    server.listen(settings.port, () => {
-        console.log(`Server listening on port ${settings.port}`);
+    server.listen(settings?.port, () => {
```
Settings is guaranteed to exist because the file is created with defaults if it does not.

subtosharki replied:
the ?. does not really mean anything, its just to make typescript happy. it has to be there because the settings variable can be null

whitigol commented on apps/api/src/index.ts:
```diff
@@ -106,14 +106,14 @@ async function initAPI() {
 
     // #region Start Version Check
-    StartVersionCheck();
+    await StartVersionCheck();
```
`StartVersionCheck` shouldn't need to be awaited here since it's async properties are only used internally in the function, since it runs on a loop.

subtosharki replied:
its being awaited because of the GET request happening in it

whitigol commented on apps/api/src/index.ts:
```diff
@@ -106,14 +106,14 @@ async function initAPI() {
-        console.log(`Server listening on port ${settings.port}`);
+    server.listen(settings?.port, () => {
+        console.log(`Server listening on port ${settings?.port}`);
         openInBrowser();
```
```suggestion
```
This can be removed, it's no longer used.

subtosharki replied:
the ?. does not really mean anything, its just to make typescript happy. it has to be there because the settings variable can be null

whitigol commented on apps/api/src/index.ts:
```diff
@@ -126,6 +126,6 @@ initAPI();
 
 function openInBrowser() {
     if (process.env.NODE_ENV === "development") {
-        exec(`start http://localhost:${settings.port}`);
+        exec(`start http://localhost:${settings?.port}`);
     }
 }
```
```suggestion
```
Can be removed, no longer used.

subtosharki replied:
the ?. does not really mean anything, its just to make typescript happy. it has to be there because the settings variable can be null

whitigol commented on apps/api/src/sockets/save-env.ts:
```diff
@@ -1,18 +1,15 @@
-type SaveEnvData = {
-    id: string;
-    env: Env;
-};
+
```
```suggestion
```
Unneeded double spacing.

whitigol reviewed (changes requested):
Thanks for the PR! I've left some change requests, but overall it looks good, thanks for the contribution!

subtosharki committed:
Apply suggestions from code review

Co-authored-by: Whitigol <36452182+WhitigolProd@users.noreply.github.com>

subtosharki committed:
Update apps/api/src/index.ts

Co-authored-by: Whitigol <36452182+WhitigolProd@users.noreply.github.com>

subtosharki committed:
- added explicit type import
- added missing alt tags
- simplified imports
- removed unused imports
- removed unused code
- added type for settings in index.ts instead of it being any
- in load-instances.ts switched from forEach to for-of because each promise returned was ignored
- added await where needed
- added ide specific folders to .gitignore
- simplified code in areas

whitigol commented:
Thank you!
